control-plane-operator/controllers/hostedcontrolplane/manifests/manifests.go

```diff
@@ -0,0 +1,146 @@
+package manifests
+
+import (
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	kubeAPIServerServiceName      = "kube-apiserver"
+	OauthServiceName              = "oauth-openshift"
+	oauthRouteName                = "oauth"
+	vpnServiceName                = "openvpn-server"
+	openshiftAPIServerServiceName = "openshift-apiserver"
+	oauthAPIServerName            = "openshift-oauth-apiserver"
+	DefaultAPIServerPort          = 6443
+	VPNServicePort                = 1194
+)
+
+func KubeAPIServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeAPIServerServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func KubeAPIServerServicePorts(securePort int32) []corev1.ServicePort {
+	return []corev1.ServicePort{
+		{
+			Port:       securePort,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(int(securePort)),
+		},
+	}
+}
+
+func KubeAPIServerServiceSelector() map[string]string {
+	return map[string]string{"app": "kube-apiserver"}
+}
+
+func OauthServerService(hostedClusterNamespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OauthServiceName,
+			Namespace: hostedClusterNamespace,
+		},
+	}
+}
+
+func OauthServerServicePorts() []corev1.ServicePort {
```

Comment:
As mentioned elsewhere, let's keep this package scoped to names only, not reconcile logic/data